src/itemlistformaction.cpp

```diff
@@ -1508,4 +1409,139 @@ std::string ItemListFormAction::title()
 	}
 }
 
+void ItemListFormAction::handle_op_saveall() {
+	LOG(Level::INFO,
+		"ItemListFormAction: saving all items");
+
+	if (visible_items.empty()) {
+		return;
+	}
+
+	std::string directory = v->run_dirbrowser();
+
+	if (directory == "") {
+		return;
+	}
+
+	if (directory.back() != '/') {
+		directory = directory + "/";
+	}
+
+	if (visible_items.size() == 1) {
+		const std::string filename = v->get_filename_suggestion(visible_items[0].first->title());
+		const std::string fpath = directory + filename;
+
+		struct stat sbuf;
+		if (::stat(fpath.c_str(), &sbuf) != -1) {
+			std::string input_options = _("yn");
+			char c = v->confirm(strprintf::fmt(
+					_("Overwrite `%s' in `%s?' "
+					  "(y:Yes n:No)"),
+					filename, directory),
+					input_options);
+			if (!c) {
+				return;
+			}
+
+			// Check that the number of translated answers is the same as the
+			// number of answers we expect to handle. If it doesn't, just give
+			// up. That'll prevent this function from saving anything, so users
+			// will complain, and we'll ask them to update the translation.
+			// A bit lame, but it's better than mishandling the answer.
+			const auto n_options = ((std::string) "yn").length();
+			if (input_options.length() < n_options) {
+				return;
+			}
+
+			if (c == input_options.at(0)) {
+				save_article(fpath, visible_items[0].first);
+			} else if (c == input_options.at(1)) {
+				return;
+			}
+		} else {
+			// Create file since it does not exist
+			save_article(fpath, visible_items[0].first);
+		}
+	} else {
+		std::vector<std::string> filenames;
+		for (const auto &item : visible_items) {
+			auto filename = v->get_filename_suggestion(item.first->title());
```

Comment:
Technically: nope, because I won't be able to move from it then. But read on :)

At the time of refactoring, I thought I'm helping the reader by explaining what `v->get_filename_suggestion` returns, but now that I re-read the whole thing I see that filenames are mentioned plenty of times already :) I'll just call `emplace_back` on a temporary.